### DISCUSSION TITLE
How to get vsmd?

### DISCUSSION THREAD

**Discussion by XuanhongZhou** [UMEP-dev/UMEP#674] [Q&A]
Dear developer, I am using the SUEWS_BiogenCO2 part of the SUEWS model for my Biogenic carbon flux model.
Since we need to calculate the “gfunc”, we refer to the “SUBROUTINE surfaceresistance” procedure （https://suews.readthedocs.io/en/stable/_static/html/suews__phys__resist_8f95_source.html）to calculate the gfunc, but this procedure requires another variable “vsmd “. We were confused for a long time, trying to figure out how to get the value of vsmd, whether we should get it by some kind of computational procedure, or whether the value of vsmd should be given in the input file.
  By the way, We have found a way to compute vsmd in a literature, is this process appropriate?
![image](https://github.com/user-attachments/assets/a8a70168-21de-47a1-8b6a-485182a47c1e)

  Looking forward to your answer Thanks a lot!

**Comment by XuanhongZhou** [comment]
@sunt05

**Reply by sunt05** [reply]
Hi @XuanhongZhou,

That's the soil moisture deficit, calculated by SUEWS based on the water balance.

**Reply by XuanhongZhou** [reply]
Thanks! I found the calculation in https://suews.readthedocs.io/en/stable/_static/html/suews__phys__waterdist_8f95_source.html .
![image](https://github.com/user-attachments/assets/da384926-ec7d-4af3-8379-d300b854191c)
  I've tried using this code before, but have two more questions about it：
  1. In the code, “soilstorecap” seems to be an array, but in the input file “SUEWS_Soil.txt”, “soilstorecap “ is a value. We can't assign a value for each kind of ground table correspondingly, only a uniform value. Or we should provide “soilstorecap” for each type of surface in the input file.
  2. How to get “soilstore_id”? By calculation or input files?
  These questions have been bothering us for a long time and we are really looking forward to your answers! Thank you very much! @sunt05

**Reply by sunt05** [reply]
> 1. In the code, “soilstorecap” seems to be an array, but in the input file “SUEWS_Soil.txt”, “soilstorecap “ is a value. We can't assign a value for each kind of ground table correspondingly, only a uniform value. Or we should provide “soilstorecap” for each type of surface in the input file.

They should be provided for each land cover type.

> 2. How to get “soilstore_id”? By calculation or input files?

It's calculated internally, based on the water balance.
